doc/api/GVT-g_api.rst

```diff
@@ -0,0 +1,297 @@
+.. _GVT-g_api:
+
+ACRN GVT-g APIs
+###############
+
+GVT-g is Intel's open source GPU virtualization solution and is up-streamed to
+the Linux kernel. Its implementation over KVM is named KVMGT, over Xen it is
+named XenGT, and over ACRN it is named ACRN-GT. GVT-g can exports multiple
+virtual GPU (vGPU) instances for virtual machine system (VM). A VM could be
+assigned one vGPU instance. The guest OS graphic driver needs minor
+modification to drive the vGPU adapter in a VM. Every vGPU instance will adopt
+the full HW GPU’s accelerate capability for 3D render and display.
+
+In the following document, ACRN-GT refers to the glue layer between ACRN
+hypervisor and GVT-g core device model. It works as the agent of
+hypervisor-related services. It is the only layer that needs to get rewritten
+when porting GVT-g to another hypervisor. For simplicity, in the rest of this
+document, GVT is used to refer to the core device model component of GVT-g,
+specifically corresponding to ``gvt.ko`` when built as a module.
+
+Core Driver Infrastructure
+**************************
+
+This section covers core driver infrastructure API used by both the display
+and the `Graphics Execution Manager(GEM)`_ parts of `i915 driver`_.
+
+.. _Graphics Execution Manager(GEM): https://lwn.net/Articles/283798/
+
+.. _i915 driver: https://01.org/linuxgraphics/gfx-docs/drm/gpu/i915.html
+
+Intel GVT-g Guest Support(vGPU)
+===============================
+
+.. kernel-doc:: drivers/gpu/drm/i915/i915_vgpu.c
+   :doc: Intel GVT-g guest support
+
+.. kernel-doc:: drivers/gpu/drm/i915/i915_vgpu.c
+   :internal:
+
+Intel GVT-g Host Support(vGPU device model)
+===========================================
+
+.. kernel-doc:: drivers/gpu/drm/i915/intel_gvt.c
+   :doc: Intel GVT-g host support
+
+.. kernel-doc:: drivers/gpu/drm/i915/intel_gvt.c
+   :internal:
+
+
+VHM APIs called from ACRN-GT
+****************************
+
+The Virtio and Hypervisor Service Module (VHM) is a kernel module in the
+Service OS acting as a middle layer to support the device model. (See the
+:ref:`ACRN-io-mediator` introduction for details.)
+
+VHM requires an interrupt( ``vIRQ`` ) number, and exposes some APIs to external
```

Comment:
You missed some of these in your recent update:

    an interrupt (vIRQ) number